src/controllers/authenticationController.ts

```diff
@@ -151,7 +167,24 @@ export default class AuthenticationController {
           type: "INFO",
         },
       });
-      return ResponsePermanentRedirect(redirectionUrl);
+
+      return pipe(
+        isUserElegibleForIoLoginUrlScheme(spidUser.fiscalNumber),
+        B.fold(
+          () => redirectionUrl,
+          () =>
+            pipe(
+              redirectionUrl,
+              errorOrIoLoginURL,
+              E.getOrElseW((err) => {
+                throw new Error(
+                  `Invalid url | ${readableReportSimplified(err)}`
+                );
+              })
+            )
+        ),
+        ResponsePermanentRedirect
+      );
```

Comment:
Return ResponseErrorInternal instead of throwing an Error.
Also, we could refactor it by extracting a function like

   ```typescript
 return pipe(
      isUserElegibleForIoLoginUrlScheme(user.fiscal_code),
      B.fold(
        () => E.of(ResponsePermanentRedirect(urlWithToken)),
        () => getInternalErrorOrIoLoginRedirect(urlWithToken)
      ),
      E.toUnion
    );
```

Reply:
nice suggestion.

Reply:
I applied it in https://github.com/pagopa/io-backend/pull/1012/commits/44f7df4397351a7e34ae3a9c1d36efa36744e006